packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -360,7 +362,8 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     let {
       value,
       allowFreeform,
-      autoComplete
+      autoComplete,
+      overrideOptionInput
```

Comment:
I took a look at this in relation to what you described and I think the fix is rather simple (and you don't need any of the added props).

If you look at the first if check inside of  `_getVisibleValue` : `if (!(isOpen && currentPendingIndexValid) && value) {` it needs to be updated to this: `if (!(isOpen && currentPendingIndexValid) && (value && !currentPendingValue)) {`. Originally this code took the stance that if there was a value, were going to always use that. The issue is that when you are in the process of editing the value (e.g. there's a `currentPendingValue`) it would get ignored and the input would always be `value`. The fix is to loosen the condition for when we use `value` to be `value` is defined and we don't have a `currentPendingValue`.

With this fix I was able to achieve the scenarios you described (including when a value and a selectedKey are both provided).